build.gradle

```diff
@@ -18,128 +18,171 @@ plugins {
+            // Currently that is developed in Kotlin 1.7 so picking the same version ensures compatibility with that.
+
+            // Specify the version of Java to target.
+            jvmTarget = javaVersion
+
+            // Needed for reflection to work correctly.
+            javaParameters = true
+
+            // -Xjvm-default determines how Kotlin supports default methods.
+            // JetBrains currently recommends developers use -Xjvm-default=all
+            // https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/
+            freeCompilerArgs += [
+                    "-Xjvm-default=all"
+            ]
+        }
     }
-}
 
-repositories {
-    // All dependencies are held in Maven Central
-    mavenCentral()
+    repositories {
+        // All dependencies are held in Maven Central
+        mavenCentral()
+        maven {
+            url = "$artifactoryContextUrl/"
+        }
+
+
+        /*
+    // Internal artifactory repos for dev
+    // R3 Internal repositories
+    // Repository the provides kotlin-stdlib-jdk8-osgi created by R3.
+    // Final location to be decided.
     maven {
-        url = "$artifactoryContextUrl/"
+        url = "$artifactoryContextUrl/corda-dependencies"
+    }
+    // Repository provides Corda 5 binaries that implement Corda-API.
+    // These will be made publicly available.
+    // Final location to be decided.
+    // Repository subject to change
+    maven {
+        url = "$artifactoryContextUrl/corda-os-maven"
+        credentials {
+            username = findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+            password = findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+        }
+    }
+    // Provides the combined-worker Jars
+    // These will be made publicly available.
+    // Final location to be decided.
+    maven {
+        url = "$artifactoryContextUrl/corda-ent-maven-unstable-local"
+        credentials {
+            username = findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+            password = findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+        }
+    }
+    maven {
+        url = "$artifactoryContextUrl/corda-ent-maven"
+        credentials {
+            username = findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+            password = findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+        }
+    }
+    maven {
+        url = "$artifactoryContextUrl/engineering-tools-maven-unstable"
+        credentials {
+            username = findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+            password = findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+        }
+    }
+
+     */

```

Comment:
We should remove commented out code though its useful as it makes switching to internal builds for testing. We may not want in the the release branch even if we are only doing RCs.

Reply:
This feels like a problem with our branching strategy that the internal and external versions can't live side by side
 We'll leave it in for now and sort it out prior to release.